extensions/lastpass/src/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { Action, ActionPanel, getPreferenceValues, Icon, List, showToast, Toast } from "@raycast/api";
+import { getFavicon } from "@raycast/utils";
+import { useEffect, useState } from "react";
+import { AccountDetail, UnknownError, CommandNotFoundError } from "./components";
+import { lastPass } from "./utils/cli";
+
+interface Preferences {
+  email: string;
+  password: string;
+}
+
+const faviconForDomain = (url: string) => getFavicon(url, { fallback: Icon.Key });
+
+const ErrorDetails = (args: { error: Error }) => {
+  if (args.error.message.includes("command not found")) {
+    showToast({
+      style: Toast.Style.Failure,
+      title: "LastPass CLI not found",
+      message: args.error.message,
+    });
+    return <CommandNotFoundError />;
+  } else {
+    showToast({
+      style: Toast.Style.Failure,
+      title: "Something went wrong",
+      message: args.error.message,
+    });
+    return <UnknownError error={args.error} />;
+  }
+};
+
+export default function Command() {
+  const { email, password } = getPreferenceValues<Preferences>();
+  const api = lastPass(email, password);
+  const [accounts, setAccounts] = useState<
+    { id: string; name: string; username: string; password: string; url: string }[]
+  >([]);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const isLogged = await api.isLogged();
+        if (!isLogged) {
+          await api.login();
+        }
+
+        const account = await api.list();
+        setAccounts(account);
+      } catch (error) {
+        if (error instanceof Error) {
+          setError(error);
+        }
+      }
+    })();
+  }, []);
+
+  return error ? (
+    <ErrorDetails error={error} />
+  ) : (
+    <List
+      isLoading={accounts.length === 0}
```

Comment:
this isn't a great idea, as it'll lead to infinite loading if users have no accounts in LastPass :D an edge case, I know, but I just ran into (as I don't use LastPass, I needed to create a new account 😸)

Reply:
true, thanks!
i will add one more details view for such case

Reply:
nice! btw, you can also use an [EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) for similar results.